test/e2e/clusterctl_upgrade_test.go

```diff
@@ -144,8 +144,8 @@ var _ = Describe("When testing clusterctl upgrades (v1.6=>current)", func() {
 			InitWithBinary:            fmt.Sprintf(clusterctlDownloadURL, stableRelease),
 			InitWithProvidersContract: "v1beta1",
 			//  Note: Both InitWithKubernetesVersion and WorkloadKubernetesVersion should be the highest mgmt cluster version supported by the source Cluster API version.
-			InitWithKubernetesVersion: "v1.28.0",
-			WorkloadKubernetesVersion: "v1.28.0",
+			InitWithKubernetesVersion: "v1.29.0",
+			WorkloadKubernetesVersion: "v1.29.0",
```

Comment:
This might require cherry-picking the kind_provider change first (?)

Reply:
cherry-pick and a release